### ISSUE TITLE
Suggestion: Swift Update

### ISSUE THREAD

**Issue by Craytor** [slavavdovichenko/MediaLibDemos3x#49]
Can you make another demo inside of this projected adapted to the Swift language? I'm having issues/confused on how to get started.

Thanks!

**Comment by BraydenGirard** [comment]
I am also working on a Swift implementation of this. To get it working I had to create an Objective C and then integrate my Swift code with the Objective C project, doing it the other way around would not work for me. 

Currently I am having an issue with auto layout constraints being modified in the background, seems to be an iOS 8/9 issue. The error is as follows:

"This application is modifying the autolayout engine from a background thread, which can lead to engine corruption and weird crashes.  This will cause an exception in a future release."

Not sure if anyone else has encountered this or how to fix it.

**Comment by Craytor** [comment]
@BraydenGirard Is your version open source? Yesterday when I was playing with the project, I thought I would change it to auto size, and I got the same message.

**Comment by BraydenGirard** [comment]
It is currently private but I can open source it no problem, would you be interested in taking a look ?

**Comment by Craytor** [comment]
Yeah, I would. I have no idea how to go about the swift part. I'm wondering if the preview view of the stream could be causing the issue. Also, I've found a project that's already in swift called VideoCore (https://github.com/jgh-/VideoCore).

**Comment by BraydenGirard** [comment]
Here is a link to my project

https://github.com/BraydenGirard/streamfit.tv_ios

Thanks

**Comment by Craytor** [comment]
Oh, is yours only for viewing the streams? Or does it broadcast as well?

**Comment by BraydenGirard** [comment]
It is supposed to be broadcasting, I must have taken the wrong target that would make sense now haha

**Comment by slavavdovichenko** [comment]
Hi guys,

you can try the swift sample based on Backendless SDK (which uses MediaLibiOS lib):
https://github.com/Backendless/iOS-Samples
(see VideoService target - https://github.com/Backendless/iOS-Samples/tree/master/samples/swift)

Backendless SDK is open source - https://github.com/Backendless/ios-SDK/tree/master/SDK - so you can investigate Media group for example how to use MediaLibiOS.

**Comment by slavavdovichenko** [comment]
Probably that could be helpful for you: https://backendless.com/developing-an-ios-video-recorder-and-player-on-swift-in-15-minutes/
